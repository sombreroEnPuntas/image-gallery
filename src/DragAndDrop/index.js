import React from 'react'
import PropTypes from 'prop-types'

// Library Components
import ImageDropZone from 'react-image-dropzone'

const DragAndDrop = ({ addPicture, removePicture, src }) => (
  <ImageDropZone
    anySize
    height={150}
    imageDefault={src}
    imageDeleted={removePicture}
    imagePicked={addPicture}
    // showButton
    // showDeleteButton
    style={{ width: '150px', height: '150px' }}
    width={150}
  />
)

DragAndDrop.propTypes = {
  addPicture: PropTypes.func.isRequired,
  removePicture: PropTypes.func.isRequired,
  src: PropTypes.string,
}

DragAndDrop.defaultProps = {}

export default DragAndDrop
