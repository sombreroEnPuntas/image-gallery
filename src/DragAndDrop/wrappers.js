import styled from 'styled-components'

import { mobileBreakpointInPixels } from '../data/constants'

export const Button = styled.button`
  border-radius: 50%;
  border: 0;
  padding: 0;
  height: 32px;
  width: 32px;

  background-color: rgba(10, 10, 10, 50%);
  color: papayawhip;
  left: 70px;
  position: relative;
  top: -160px;

  @media (min-width: ${mobileBreakpointInPixels}px) {
    top: -299px;
    left: 132px;
  }
`
