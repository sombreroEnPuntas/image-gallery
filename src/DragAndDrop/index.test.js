import React from 'react'
import { shallow } from 'enzyme'

import DragAndDrop from './'

// Mocks
import mockPictures from '../mocks/pictures'

describe('DragAndDrop', () => {
  it('renders mocks as expected for: Pictures!', () => {
    const wrapper = shallow(
      <DragAndDrop
        addPicture={jest.fn()}
        removePicture={jest.fn()}
        src={mockPictures[0].src}
      />
    )
    expect(wrapper.debug()).toMatchSnapshot()
  })

  it('rendersempty', () => {
    const wrapper = shallow(
      <DragAndDrop addPicture={jest.fn()} removePicture={jest.fn()} />
    )
    expect(wrapper.debug()).toMatchSnapshot()
  })
})
