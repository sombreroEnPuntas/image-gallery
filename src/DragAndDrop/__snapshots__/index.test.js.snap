// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DragAndDrop on desktop with pic renders empty 1`] = `
"<Fragment>
  <t anySize={true} height={300} imageDefault={[undefined]} imageDeleted={[Function: mockConstructor]} imagePicked={[Function: mockConstructor]} showButton={true} showDeleteButton={true} width={300} />
</Fragment>"
`;

exports[`DragAndDrop on desktop with pic renders mocks as expected for: Pictures! 1`] = `
"<Fragment>
  <t anySize={true} height={300} imageDefault=\\"https://source.unsplash.com/random/150x150?sig=0\\" imageDeleted={[Function: mockConstructor]} imagePicked={[Function: mockConstructor]} showButton={true} showDeleteButton={true} width={300} />
  <styled.button onClick={[Function: clickRemove]}>
    X
  </styled.button>
</Fragment>"
`;

exports[`DragAndDrop on mobile with mo pic renders empty 1`] = `
"<Fragment>
  <t anySize={true} height={150} imageDefault={[undefined]} imageDeleted={[Function: mockConstructor]} imagePicked={[Function: mockConstructor]} showButton={true} showDeleteButton={true} width={150} />
  <styled.button onClick={[Function: clickAdd]}>
    +
  </styled.button>
</Fragment>"
`;

exports[`DragAndDrop on mobile with pic renders mocks as expected for: Pictures! 1`] = `
"<Fragment>
  <t anySize={true} height={150} imageDefault=\\"https://source.unsplash.com/random/150x150?sig=0\\" imageDeleted={[Function: mockConstructor]} imagePicked={[Function: mockConstructor]} showButton={true} showDeleteButton={true} width={150} />
  <styled.button onClick={[Function: clickRemove]}>
    X
  </styled.button>
</Fragment>"
`;
